src/frontend/src/instance/otlp.rs

```diff
@@ -70,7 +71,13 @@ impl OpenTelemetryProtocolHandler for Instance {
             .check_permission(ctx.current_user(), PermissionReq::Otlp)
             .context(AuthSnafu)?;
 
-        let (requests, rows) = otlp::traces::to_grpc_insert_requests(request)?;
+        let (requests, rows) = otlp::trace::to_grpc_insert_requests(request)?;
+
+        let requests = self
+            .plugins
+            .get::<TraceTransformerRef>()
+            .as_ref()
+            .transform(requests);
```

Comment:
We can use plugin to provide alternative implementation of `to_grpc_insert_requests` so that we don't need to copy/parse encoded `InsertRequests`